jira_agile_metrics/calculators/defects.py

```diff
@@ -77,120 +90,198 @@ def run(self):
+            self.write_defects_by_environment_chart(chart_data, self.settings['defects_by_environment_chart'],x_label)
+
+        if self.settings['defects_by_team_chart']:
+            team_values = self.settings['defects_team_values']
+            for team in team_values:
+                sorted_df= chart_data.copy()
+                sorted_df.drop(sorted_df[sorted_df['team'] != team].index, inplace = True) 
+                # sc.sort_values("team", inplace = True)
+                # print(sc)
+                if sorted_df.empty:
+                    continue
+                self.write_defects_by_team_chart(sorted_df, self.settings['defects_by_team_chart'],x_label,team)
+
+    def write_defects_by_priority_chart(self, chart_data, output_file,x_label):
         window = self.settings['defects_window']
         priority_values = self.settings['defects_priority_values']
 
-        breakdown = breakdown_by_month(chart_data, 'created', 'resolved', 'key', 'priority', priority_values)
-        
+
+        if x_label !='Month':
+            breakdown = chart_data.pivot_table(index=['fixversion'],values='key',columns=['priority'],aggfunc='count')
+        else:
+            breakdown = breakdown_by_month(chart_data, 'created', 'resolved', 'key', 'priority', priority_values)
+
+
+
         if window:
             breakdown = breakdown[-window:]
-        
+
         if len(breakdown.index) == 0 or len(breakdown.columns) == 0:
             logger.warning("Cannot draw defects by priority chart with zero items")
             return
 
         fig, ax = plt.subplots()
-        
+
         breakdown.plot.bar(ax=ax, stacked=True)
-        
+
         if self.settings['defects_by_priority_chart_title']:
             ax.set_title(self.settings['defects_by_priority_chart_title'])
 
         ax.legend(loc='center left', bbox_to_anchor=(1, 0.5))
-        ax.set_xlabel("Month", labelpad=20)
-        ax.set_ylabel("Number of items", labelpad=10)
+        ax.set_xlabel(x_label, labelpad=20)
+        ax.set_ylabel("Number of Defects", labelpad=10)
+
+        if x_label =='Month':
+            labels = [d.strftime("%b %y") for d in breakdown.index]
+            ax.set_xticklabels(labels, rotation=90, size='small')
+            ax.set_xlabel("Month", labelpad=20)
 
-        labels = [d.strftime("%b %y") for d in breakdown.index]
-        ax.set_xticklabels(labels, rotation=90, size='small')
 
-        set_chart_style()
+        # set_chart_style()
 
         # Write file
         logger.info("Writing defects by priority chart to %s", output_file)
         fig.savefig(output_file, bbox_inches='tight', dpi=300)
         plt.close(fig)
-    
-    def write_defects_by_type_chart(self, chart_data, output_file):
+
+    def write_defects_by_type_chart(self, chart_data, output_file,x_label):
         window = self.settings['defects_window']
         type_values = self.settings['defects_type_values']
+       
+        if x_label !='Month':
```

Comment:
Same point as above re comparator

Reply:
config is initialised in config.py  as below
`'defects_fixversion_label': 'Month',`
So if it is overridden then only the Sprint level charting code will be executed.